File: CS/Grid_BottomNewItemRow/GridDescendant/GridControlDescendant.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Grid_BottomNewItemRow
{
    public class GridControlDescendant : GridControl
    {      
        protected override BaseView CreateDefaultView()
        {
            return CreateView("GridViewDescendant");
        }
        protected override void RegisterAvailableViewsCore(InfoCollection collection)
        {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new GridViewInfoRegistratorDescendant());
        }


    }
}

File: VB/Grid_BottomNewItemRow/GridDescendant/GridControlDescendant.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Grid_BottomNewItemRow
	Public Class GridControlDescendant
		Inherits GridControl

		Protected Overrides Function CreateDefaultView() As BaseView
			Return CreateView("GridViewDescendant")
		End Function
		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New GridViewInfoRegistratorDescendant())
		End Sub


	End Class
End Namespace
